test/test_optim.py

```diff
@@ -603,6 +646,31 @@ def test_new_pattern_no_warning_with_arg(self):
                 scheduler.step(e)
             self.assertTrue(len(ws) == 0, "No warning should be raised")
 
+    def test_new_pattern_no_warning_with_overriden_optim_step(self):
```

Comment:
@mcarilli Could you confirm that this test case accurately captures the wrapping that APEX is doing on step?

Reply:
Looks reasonable to me.  For iterations that Amp chooses to skip, Amp patches optimizer.step to a new method that does not carry out an update, but instead, restores optimizer.step to the original method.